nautobot/ipam/views.py

```diff
@@ -52,17 +61,133 @@ def get_extra_context(self, request, instance):
         context = super().get_extra_context(request, instance)
 
         if self.action == "retrieve":
-            vrfs = instance.vrfs.restrict(request.user, "view")
-            vrf_table = tables.VRFTable(vrfs, orderable=False)
-            vrf_table.exclude = ("namespace",)
-            context["vrf_table"] = vrf_table
-
-            prefixes = instance.prefixes.restrict(request.user, "view")
-            prefix_count = prefixes.count()
-            prefix_table = tables.PrefixTable(prefixes)
-            prefix_table.exclude = ("namespace",)
-            context["prefix_count"] = prefix_count
-            context["prefix_table"] = prefix_table
+            context.update(get_namespace_related_counts(instance, request))
+
+        return context
+
+
+class NamespaceIPAddressesView(generic.ObjectView):
+    queryset = Namespace.objects.all()
+    template_name = "ipam/namespace_ipaddresses.html"
+
+    def get_extra_context(self, request, instance):
+        # Find all IPAddresses belonging to this Namespace
+        ip_addresses = instance.ip_addresses.restrict(request.user, "view").select_related("status")
+
+        ip_address_table = tables.IPAddressTable(ip_addresses)
+        if request.user.has_perm("ipam.change_ipaddress") or request.user.has_perm("ipam.delete_ipaddress"):
+            ip_address_table.columns.show("pk")
+
+        ip_address_table.exclude = ("namespace",)
+
+        paginate = {
+            "paginator_class": EnhancedPaginator,
+            "per_page": get_paginate_count(request),
+        }
+        RequestConfig(request, paginate).configure(ip_address_table)
+
+        # Compile permissions list for rendering the object table
+        permissions = {
+            "add": request.user.has_perm("ipam.add_ipaddress"),
+            "change": request.user.has_perm("ipam.change_ipaddress"),
+            "delete": request.user.has_perm("ipam.delete_ipaddress"),
+        }
+        bulk_querystring = f"namespace={instance.id}"
+
+        context = super().get_extra_context(request, instance)
+        context.update(
+            {
+                "ip_address_table": ip_address_table,
+                "permissions": permissions,
+                "bulk_querystring": bulk_querystring,
+                "active_tab": "ip-addresses",
+            }
+        )
+        context.update(get_namespace_related_counts(instance, request))
```

Comment:
Why isn't this being performed by `super().get_extra_context`? Is it because of the `if self.action == "retrieve":` line in the super method?

Reply:
Because these views don't inherit from the parent.